### ISSUE TITLE
Docker server not honoring "--iptables=false"

### ISSUE THREAD

**Issue by maticmeznar** [docker/docker#19063]
It seems Docker is still managing iptables even with --iptables=false option. Specifically, it seems to delete my DOCKER chain and all references to it. I created this DOCKER chain with my firewall script, not Docker server, yet it still gets removed.

This is my nat table before starting Docker server:
```
$ iptables -S -t nat
-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
-N DOCKER
-A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
-A POSTROUTING ! -s X.X.X.X/32 -o ppp0 -j SNAT --to-source X.X.X.X --random
-A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
```
After starting Docker server, all rules in nat table containing string DOCKER seem to be removed!
```
$ systemctl start docker
$ iptables -S -t nat
-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
-A POSTROUTING ! -s X.X.X.X/32 -o ppp0 -j SNAT --to-source X.X.X.X --random
-A POSTROUTING -s 172.17.0.0/16 ! -o docker0 -j MASQUERADE
```
Docker server is configured to not touch iptables!
```
$ cat /etc/sysconfig/docker |grep ipt
OPTIONS='--selinux-enabled --userland-proxy=false --iptables=false'
```
```
$ docker version
Client:
 Version:      1.8.2-el7.centos
 API version:  1.20
 Package Version: docker-1.8.2-10.el7.centos.x86_64
 Go version:   go1.4.2
 Git commit:   a01dc02/1.8.2
 Built:        
 OS/Arch:      linux/amd64

Server:
 Version:      1.8.2-el7.centos
 API version:  1.20
 Package Version: 
 Go version:   go1.4.2
 Git commit:   a01dc02/1.8.2
 Built:        
 OS/Arch:      linux/amd64
```
OS: Centos 7 (with all updates)

**Comment by lenovouser** [comment]
I have the same issue

**Comment by phenomax** [comment]
Same problem here :(

**Comment by maticmeznar** [comment]
```
$ docker info
Containers: 3
Images: 126
Storage Driver: devicemapper
 Pool Name: vg--akula-docker--pool
 Pool Blocksize: 524.3 kB
 Backing Filesystem: extfs
 Data file: 
 Metadata file: 
 Data Space Used: 3.079 GB
 Data Space Total: 7.629 GB
 Data Space Available: 4.55 GB
 Metadata Space Used: 1.413 MB
 Metadata Space Total: 163.6 MB
 Metadata Space Available: 162.2 MB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Library Version: 1.02.107-RHEL7 (2015-10-14)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-327.3.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 2
Total Memory: 2.438 GiB
Name: griad.fdsfs.fdsg.ds
ID: GFQ3:LXJ4:KX3R:HOMJ:L6RD:5QY6:ZQRI:YZTF:V6I2:KLP4:L3VW:5A2A
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```
Is anyone reading this? Do you need anything else?

**Comment by lenovouser** [comment]
This is still an issue for some reason.

`/etc/default/docker`:
```
# Docker Upstart and SysVinit configuration file

# Customize location of Docker binary (especially for development testing).
#DOCKER="/usr/local/bin/docker"

# Use DOCKER_OPTS to modify the daemon startup options.
DOCKER_OPTS="--dns 8.8.8.8 --dns 8.8.4.4 --iptables=false"

# If you need Docker to use an HTTP proxy, it can also be specified here.
#export http_proxy="http://127.0.0.1:3128/"

# This is also a handy place to tweak where Docker's temporary files go.
#export TMPDIR="/mnt/bigdrive/docker-tmp"
```
I delete the docker rules with:

`iptables -F DOCKER`

then I do

`service docker restart`

when I then look into the IPTables rules with `iptables -L` the docker rules show up again.

I am using docker `version 1.10.2, build c3959b1`

**Comment by thaJeztah** [comment]
@lenovouser what distro and version are you running on? Note that that file is only for distro's using upstart, not for systemd

**Comment by lenovouser** [comment]
@thaJeztah I am using Debian 8. When I put this:

```
[Unit]
Description=Docker Application Container Engine
Documentation=http://docs.docker.com
After=network.target docker.socket
Requires=docker.socket

[Service]
EnvironmentFile=-/etc/default/docker
ExecStart=/usr/bin/docker -d $DOCKER_OPTS -H fd://
MountFlags=slave
LimitNOFILE=1048576
LimitNPROC=1048576
LimitCORE=infinity

[Install]
WantedBy=multi-user.target
```
in my `/etc/systemd/system/docker.service.d/docker.service`, remove the iptables rules, reload systemd and then restart docker it still adds the rules though.

**Comment by thaJeztah** [comment]
@lenovouser do you have a drop-in file (https://docs.docker.com/engine/admin/systemd/#custom-docker-daemon-options) somewhere? because `docker -d` is replaced by `docker daemon`, so should not work with docker 1.10 (see https://github.com/docker/docker/blob/master/docs/deprecated.md#old-command-line-options), so it wondering if your settings are overridden somewhere
